pkg/kubelet/kubelet.go

```diff
@@ -1764,6 +1767,32 @@ func (kl *Kubelet) pastActiveDeadline(pod *api.Pod) bool {
 	return false
 }
 
+// Get pods which should be resynchronized. Move the logic here so as to make it testable and make the main sync loop
+// simple and clear.
+// Currently, the following pod should be resynchronized:
+//   * pod whose work is ready.
+//   * pod past the active deadline.
+func (kl *Kubelet) getPodsToSync() []*api.Pod {
+	allPods := kl.podManager.GetPods()
+	podUIDs := kl.workQueue.GetWork()
```

Comment:
Might be easier to build a set out of the list, e.g., `util.set.NewString()`, then check whether a pod uid is in the set without having to use a for loop.